crates/l3/common/src/lib.rs

```diff
@@ -0,0 +1,35 @@
+pub mod errors;
+use starknet_accounts::{Account, Call, Execution, SingleOwnerAccount};
+use starknet_ff::FieldElement;
+use starknet_providers::jsonrpc::{HttpTransport, JsonRpcClient};
+use starknet_providers::Provider;
+use starknet_signers::{LocalWallet, SigningKey};
+use std::sync::Arc;
+
+use starknet_core::utils::get_selector_from_name;
+
+pub type LocalWalletSignerMiddleware =
+    SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>;
+pub const NO_CONSTRUCTOR_ARG: () = ();
+
+pub trait StarknetContractClient {
```

Comment:
there are two `StarknetContractClient` currently, one here and the other in the utils folder

Reply:
isn't it the same thing, do we need both?

Reply:
yes, one is using middleware from ethers library and other from starknetone

Reply:
maybe we want to implement it for `StarknetCoreContractClient`. we can call the one in utils as `EthereumContractClient`

Reply:
sure, have left a todo, will change when doing restructuring of ethereum interfaces

Reply:
let's restructure in this PR itself? ig we just need to move to a new folder right, or am I missing something

Reply:
next step would be PRs on dependent repos which is only Madara for now I guess

Reply:
done